### DISCUSSION TITLE
DateTime format 'E' and 'e' shortcuts for a popular European datetime format style dd/MM/yyyy

### DISCUSSION THREAD

**Discussion by avertiy** [dotnet/csharplang#8097] [Q&A]
I wondered why C# does not yet provide a special shortcut for a popular European date style format

G or g format modifiers format date time in US-style (I know it depends on culture settings but in most cases we use G shortcut): 
`dateTime.ToString("G") is equivalent of dateTime.ToString("MM/dd/yyyy HH:mm:ss") ` 
`dateTime.ToString("g") is equivalent of dateTime.ToString("MM/dd/yyyy HH:mm") `

why not to add an 'E' shortcut for European date style e.g. 

`dateTime.ToString("E")  to be equivalent of dateTime.ToString("dd/MM/yyyy HH:mm:ss") `
`dateTime.ToString("e")  to be equivalent of dateTime.ToString("dd/MM/yyyy HH:mm") `

**Comment by colejohnson66** [comment]
Format codes are not C#, but the runtime: [@dotnet/runtime](https://github.com/dotnet/runtime)

**Comment by huoyaoyuan** [comment]
> G or g format modifiers format date time in US-style

It formats in the "general" style of current culture. https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#table-of-format-specifiers

When a European culture is supplied or set as the default culture, it will format as European style.

> provide a special shortcut

The culture-aware formatting strings shouldn't be treated as a "shortcut". It's just follows current culture.
**A common mistake is to use culture-aware formatting for persisted storage.** They should only be used for displaying.
